packages/core/src/shared/env/resolveEnv.ts

```diff
@@ -114,7 +114,7 @@ export async function getSpawnEnv(
 export async function mergeResolvedShellPath(env: IProcessEnvironment): Promise<typeof process.env> {
     const shellEnv = await getResolvedShellEnv(env)
     // resolve failed or doesn't need to resolve
-    if (!shellEnv) {
+    if (!shellEnv || Object.keys(shellEnv).length === 0) {
```

Comment:
would it improve the interface if `getResolvedShellEnv` did this check and returned undefined?

Reply:
Yes that make sense. I'll change that shortly